packages/create-cloudflare/src/templateMap.ts

```diff
@@ -1,53 +1,368 @@
+	// to handle config version skew between different versions of c3
+	configVersion: number;
+	// How this template is referred to internally and keyed in lookup maps
+	id: string;
+	// How this template is presented to the user
+	displayName: string;
+	platform: "workers" | "pages";
 	hidden?: boolean;
+	languages?: string[];
+	copyFiles?: StaticFileMap | VariantInfo;
+
+	generate?: (ctx: C3Context) => Promise<void>;
+	configure?: (ctx: C3Context) => Promise<void>;
+
+	transformPackageJson?: (
+		pkgJson: Record<string, string | object>
+	) => Promise<Record<string, string | object>>;
+
+	path?: string;
+
+	testFlags?: string[];
+	compatibilityFlags?: string[];
+	deployCommand?: string[];
+	devCommand?: string[];
 };
 
-export const templateMap: Record<string, TemplateConfig> = {
-	"hello-world": {
-		label: `"Hello World" Worker`,
-		handler: runWorkersGenerator,
-	},
-	webFramework: {
-		label: "Website or web app",
-		handler: runPagesGenerator,
-	},
-	common: {
-		label: "Example router & proxy Worker",
-		handler: runWorkersGenerator,
-	},
-	scheduled: {
-		label: "Scheduled Worker (Cron Trigger)",
-		handler: (args) => runWorkersGenerator({ ...args, deploy: false }),
-	},
-	queues: {
-		label: "Queue consumer & producer Worker",
-		handler: (args) => runWorkersGenerator({ ...args, deploy: false }),
-	},
-	chatgptPlugin: {
-		label: `ChatGPT plugin`,
-		handler: (args) =>
-			runWorkersGenerator({
-				...args,
-				ts: true,
-			}),
-	},
-	openapi: {
-		label: `OpenAPI 3.1`,
-		handler: (args) =>
-			runWorkersGenerator({
-				...args,
-				ts: true,
-			}),
-	},
-	"pre-existing": {
-		label: "Pre-existing Worker (from Dashboard)",
-		handler: runWorkersGenerator,
-		hidden: true,
-	},
+// A template can have a number of variants, usually js/ts
+type VariantInfo = {
+	path: string;
+};
+
+type StaticFileMap = Record<string, VariantInfo>;
+
+export type FrameworkMap = Awaited<ReturnType<typeof getFrameworkMap>>;
+export type FrameworkName = keyof FrameworkMap;
+
+export const getFrameworkMap = async () => ({
+	angular: (await import("../templates/angular/c3")).default,
+	astro: (await import("../templates/astro/c3")).default,
+	docusaurus: (await import("../templates/docusaurus/c3")).default,
+	gatsby: (await import("../templates/gatsby/c3")).default,
+	hono: (await import("../templates/hono/c3")).default,
+	next: (await import("../templates/next/c3")).default,
+	nuxt: (await import("../templates/nuxt/c3")).default,
+	qwik: (await import("../templates/qwik/c3")).default,
+	react: (await import("../templates/react/c3")).default,
+	remix: (await import("../templates/remix/c3")).default,
+	solid: (await import("../templates/solid/c3")).default,
+	svelte: (await import("../templates/svelte/c3")).default,
+	vue: (await import("../templates/vue/c3")).default,
+});
+
+export const getTemplateMap = async () => {
+	return {
+		"hello-world": await import("../templates/hello-world/c3.json"),
```

Comment:
This is an artifact of the previous plan where templates would require a `c3.json` config file along with their template. `.json` was better than `.ts` in this scenario since we wouldn't need to worry about transpiling and executing the config file. 

Now that we're dropping the c3.json requirement and going with a convention over configuration approach, there's no real benefit.

I'll switch things over to `.ts`